cross_compile/Dockerfile_ros2

```diff
@@ -100,14 +100,5 @@ RUN rosdep update && \
 RUN colcon mixin add cc_mixin \
     https://raw.githubusercontent.com/ros-tooling/cross_compile/master/mixins/index.yaml && \
     colcon mixin update cc_mixin
+RUN touch /opt/ros/${ROS_DISTRO}/setup.sh
```

Comment:
What is `setup.sh` is empty? How does this affect the next step?

Reply:
if it doesn't exist it gets created by this command as an empty file. if it does exist then its timestamp is updated.

if it was created as empty, the initial sourcing portion of the line below is a no-op

if it already existed, then it does its thing

Reply:
however, the directory may not have existed already, in the "no-rosdeps" use case, so i'm adding a check for that